src/status_im/chat/models/message.cljs

```diff
@@ -91,15 +91,20 @@
                                                                         (not (chat-model/bot-only-chat? db chat-id))
                                                                         (not (= constants/system from)))))))
 
+(defn confirm-messages-processed [js-obj {{:keys [web3]} :db}]
+  {:confirm-message-processed {:web3   web3
+                               :js-obj js-obj}})
+
 (defn receive
-  [{:keys [chat-id message-id] :as message} {:keys [now] :as cofx}]
+  [{:keys [chat-id message-id js-obj] :as message} {:keys [now] :as cofx}]
   (handlers-macro/merge-fx cofx
                            (chat-model/upsert-chat {:chat-id chat-id
                                                     ;; We activate a chat again on new messages
                                                     :is-active true
                                                     :timestamp now})
                            (add-received-message message)
-                           (requests-events/add-request chat-id message-id)))
+                           (requests-events/add-request chat-id message-id)
+                           (confirm-messages-processed js-obj)))
```

Comment:
yeah, it's not atomic atm. Though i would test it as is and add this property in a separate PR, most likely it will require rewriting somehow `:data-store/tx` so that callback can be passed. Other ways to implement this sound reliable for me. Anyways this looks more like very rare case which we definitely need to handle, but first version can live without this.

Reply:
ok, maybe just add a `TODO` comment so we are aware of the issue.

Reply:
added